apps/studio/components/interfaces/Storage/StorageSettings/StorageSettings.tsx

```diff
@@ -149,170 +183,261 @@ const StorageSettings = () => {
-                      label="Enable Image Transformation"
-                      description={
-                        <>
-                          Optimize and resize images on the fly.{' '}
-                          <InlineLink href="https://supabase.com/docs/guides/storage/serving/image-transformations">
-                            Learn more
-                          </InlineLink>
-                          .
-                        </>
-                      }
-                    >
-                      <FormControl_Shadcn_>
-                        <Switch
-                          size="large"
-                          disabled={isFreeTier}
-                          checked={field.value}
-                          onCheckedChange={field.onChange}
-                        />
-                      </FormControl_Shadcn_>
-                    </FormItemLayout>
+                  {isSpendCapOn && (
+                    <UpgradeToPro
+                      fullWidth
+                      buttonText="Disable Spend Cap"
+                      primaryText="Reduced max upload file size limit due to Spend Cap"
+                      secondaryText={`Disable your Spend Cap to allow file uploads of up to ${formatBytes(
+                        STORAGE_FILE_SIZE_LIMIT_MAX_BYTES_UNCAPPED
+                      )}.`}
+                      source="storageSizeLimit"
+                    />
                   )}
-                />
-              </CardContent>
 
-              {isFreeTier && (
-                <CardContent className="pt-0">
-                  <UpgradeToPro
-                    primaryText="Free Plan has a fixed upload file size limit of 50 MB."
-                    secondaryText={`Upgrade to Pro Plan for a configurable upload file size limit of ${formatBytes(
-                      STORAGE_FILE_SIZE_LIMIT_MAX_BYTES_UNCAPPED
-                    )} and unlock image transformations.`}
-                    source="storageSizeLimit"
-                  />
-                </CardContent>
-              )}
-              {isSpendCapOn && (
-                <CardContent className="pt-0">
-                  <UpgradeToPro
-                    buttonText="Disable Spend Cap"
-                    primaryText="Reduced max upload file size limit due to Spend Cap"
-                    secondaryText={`Disable your Spend Cap to allow file uploads of up to ${formatBytes(
-                      STORAGE_FILE_SIZE_LIMIT_MAX_BYTES_UNCAPPED
-                    )}.`}
-                    source="storageSizeLimit"
-                  />
-                </CardContent>
-              )}
-
-              {!canUpdateStorageSettings && (
-                <CardContent className="pt-0">
-                  <p className="text-sm text-foreground-light">
-                    You need additional permissions to update storage settings
-                  </p>
-                </CardContent>
-              )}
+                  {!canUpdateStorageSettings && (
```

Comment:
Just a feedback here: Feels like the "Need additional perms" message to update settings takes up a lot of space
Presuming that we'll be figuring out a pattern for this sort of state - wondering if we should hide the CardFooter?
<img width="1087" height="287" alt="image" src="https://github.com/user-attachments/assets/f0bb9948-3dd3-48b1-b761-8c21cb72bb39" />